library/std/src/sys/sync/rwlock/queue.rs

```diff
@@ -116,23 +116,26 @@ use crate::sync::atomic::Ordering::{AcqRel, Acquire, Relaxed, Release};
 use crate::sync::atomic::{AtomicBool, AtomicPtr};
 use crate::thread::{self, Thread, ThreadId};
 
-// Locking uses exponential backoff. `SPIN_COUNT` indicates how many times the
-// locking operation will be retried.
-// `spin_loop` will be called `2.pow(SPIN_COUNT) - 1` times.
-const SPIN_COUNT: usize = 7;
-
-type State = *mut ();
+/// The atomic lock state.
 type AtomicState = AtomicPtr<()>;
+/// The inner lock state.
+type State = *mut ();
 
 const UNLOCKED: State = without_provenance_mut(0);
 const LOCKED: usize = 1;
 const QUEUED: usize = 2;
 const QUEUE_LOCKED: usize = 4;
-const DOWNGRADE: usize = 8;
+const DOWNGRADED: usize = 8;
 const SINGLE: usize = 16;
```

Comment:
So I think that this might be better:

```rust
const UNLOCKED: State = without_provenance_mut(0);
const LOCKED: usize = 1 << 0;
const QUEUED: usize = 1 << 1;
const QUEUE_LOCKED: usize = 1 << 2;
const DOWNGRADED: usize = 1 << 3;
const SINGLE: usize =  1 << 4;
const STATE: usize = DOWNGRADED | QUEUE_LOCKED | QUEUED | LOCKED;
const MASK: usize = !STATE;
```

Over this:
```rust
const UNLOCKED: State = without_provenance_mut(0);
const LOCKED: usize = 0b0001;
const QUEUED: usize = 0b0010;
const QUEUE_LOCKED: usize = 0b0100;
const DOWNGRADED: usize = 0b1000;
const SINGLE: usize =  0b10000;
const STATE: usize = DOWNGRADED | QUEUE_LOCKED | QUEUED | LOCKED;
const MASK: usize = !STATE;
```

If it was just 4 bit flags then it would probably be fine, but because we need 16 as well that last one has 5 binary digits instead of 4.

Reply:
Works for me! Just something so so anyone reading doesn't need to convert to binary in their head.